harness/determined/cli/_util.py

```diff
@@ -81,6 +83,15 @@ def make_pagination_args(
 default_pagination_args = make_pagination_args()
 
 
+def login_sdk_client(func: Callable[[argparse.Namespace], Any]) -> Callable[..., Any]:
```

Comment:
currently in master, if you are not logged in and you try to do e.g. `det user list`, the message is: 
```
Failed to list users: Unauthenticated: Please use 'det user login <username>' for password login, or for Enterprise users logging in with an SSO provider, use 'det auth login --provider=<provider>'.
```

with this PR, the message is: 
```
Failed to list users: {"error":{"code":16,"reason":"Unauthenticated","error":"invalid credentials"}}
```

I think the previous message was somewhat useful, can we please bring that back?

Reply:
I made this change. And it's working fine in the CLI. For some reason test_logout is failing because it's not seeing this error when it should. But I don't want to stall the review because of this so asking for re-review before I figure this out. Hope that's ok.